ros_buildfarm/templates/release/rpm/mock_config.cfg.em

```diff
@@ -10,7 +18,12 @@ config_opts['use_bootstrap'] = False
 config_opts['chroot_setup_cmd'] += ' python3-rpm-macros'
 
 # Install weak dependencies to get group members
-config_opts[f'dnf_builddep_opts'] = config_opts.get(f'dnf_builddep_opts', []) + ['--setopt=install_weak_deps=True']
+package_manager = _expanded('package_manager', config_opts)
+config_opts[f'{package_manager}_builddep_opts'] = config_opts.get(f'{package_manager}_builddep_opts', []) + ['--setopt=install_weak_deps=True']
+
+# Deal with dnf -> dnf4 transition - see rpm-software-management/mock#1496
+if package_manager == 'dnf':
+    config_opts[f'dnf4_builddep_opts'] = config_opts.get(f'dnf4_builddep_opts', []) + ['--setopt=install_weak_deps=True']
```

Comment:
Is this needed on top of L22 because `config_opts['dnf4_builddep_opts']` is checked in addition to / instead of `config_opts['dnf_builddep_opts']` in some situations? If so, I think @azeey's suggestion also makes sense.

Reply:
The transition in mock is pretty muddy. The situations I'm aware of are:
1. There is only `dnf` and `dnf_builddep_opts`
2. package_manager is `dnf`, which is an alias for `dnf4` and `dnf4_builddep_opts` is what we want set
3. package_manager is now `dnf4` and we want `dnf4_builddep_opts` as before

TBH it doesn't really hurt to drop the conditional and just set the config even if it won't be used, but I think may help in future forensics to understand the purpose of the duplicated code.